libs/designsystem/src/lib/components/chip/chip.component.integration.spec.ts

```diff
@@ -0,0 +1,110 @@
+import { createHostFactory, SpectatorHost } from '@ngneat/spectator';
+
+import { DesignTokenHelper } from '../../helpers';
+import { TestHelper } from '../../testing/test-helper';
+import { BadgeComponent } from '../badge/badge.component';
+import { ChipComponent } from '../chip/chip.component';
+import { SegmentItem } from '../segmented-control/segment-item';
+import {
+  SegmentedControlComponent,
+  SegmentedControlMode,
+} from '../segmented-control/segmented-control.component';
+
+const getColor = DesignTokenHelper.getColor;
+
+describe('ChipComponent in SegmentedControlComponent', () => {
+  let spectator: SpectatorHost<SegmentedControlComponent>;
+  let items: SegmentItem[] = [
+    {
+      text: 'Chip-1',
+      id: '1',
+    },
+  ];
+
+  const createHost = createHostFactory({
+    component: SegmentedControlComponent,
+    declarations: [ChipComponent, BadgeComponent],
+    imports: [TestHelper.ionicModuleForTest],
+  });
+
+  describe("when mode is 'compactChip'", () => {
+    let chip: Element;
+
+    beforeEach(() => {
+      spectator = createHost(
+        `<kirby-segmented-control [mode]="mode" [items]="items" selectedIndex="1">
+         </kirby-segmented-control>`,
+        {
+          hostProps: {
+            items: items,
+            mode: SegmentedControlMode.compactChip,
+          },
+        }
+      );
+      chip = spectator.queryHostAll('kirby-chip')[0];
```

Comment:
Consider a small refactoring to `chip = spectator.queryHost('kirby-chip');`, as it will return the first matching element anyway. This way you can also declare chip as `let chip: HTMLElement;` on line 31, for specificity.

Reply:
Good point, done!